pkg/controllers/workgenerator/controller.go

```diff
@@ -220,6 +221,33 @@ func (r *Reconciler) Reconcile(ctx context.Context, req controllerruntime.Reques
 	return controllerruntime.Result{}, syncErr
 }
 
+// updateBindingStatusWIthRetry sends the update request to API server with retry.
+func (r *Reconciler) updateBindingStatusWithRetry(ctx context.Context, resourceBinding *fleetv1beta1.ClusterResourceBinding) error {
+	// Retry only for specific errors or conditions
+	errAfterRetries := retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
This is an extra read without even trying to update. Ideally, we should update first and then retry on failure.

Reply:
Having everything wrapped inside was suggested [earlier](https://github.com/Azure/fleet/pull/906#discussion_r1733295068). Also wrapped in [here](https://github.com/Azure/fleet/blob/de6689a41f382fdbeae044cce9559c175c799d1b/pkg/controllers/workgenerator/controller.go#L275).

Reply:
@michaelawyu any thoughts?

Reply:
Changed it to your suggestion.